nutanix/data_source_nutanix_foundation_central_imaged_node_details.go

```diff
@@ -0,0 +1,285 @@
+				Computed: true,
+				Optional: true,
+			},
+			"model": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"node_position": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"node_serial": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"node_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"node_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"object_version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+				Optional: true,
+			},
+			"latest_hb_ts_list": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceFCImagedNodeDetailsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).FoundationCentral
+	req := fc.ImagedNodeDetailsInput{}
+
+	nodeUUID, ok := d.GetOk("imaged_node_uuid")
+	if !ok {
+		return diag.Errorf("please provide the imaged_node_uuid")
+	}
+	req.ImagedNodeUUID = nodeUUID.(string)
+
+	res, err := conn.Service.GetImagedNode(ctx, req.ImagedNodeUUID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("aos_version", (res.AosVersion)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("api_key_uuid", (res.APIKeyUUID)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("available", (res.Available)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("block_serial", (res.BlockSerial)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("created_timestamp", (res.CreatedTimestamp)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("current_time", (res.CurrentTime)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("cvm_gateway", (res.CvmGateway)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("cvm_ip", (res.CvmIP)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("cvm_ipv6", (res.CvmIpv6)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("cvm_netmask", (res.CvmNetmask)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("cvm_up", (res.CvmUp)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("cvm_uuid", (res.CvmUUID)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("cvm_vlan_id", (res.CvmVlanID)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("foundation_version", (res.FoundationVersion)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("hypervisor_gateway", (res.HypervisorGateway)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("hypervisor_hostname", (res.HypervisorHostname)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("hypervisor_ip", (res.HypervisorIP)); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("hypervisor_netmask", (res.HypervisorNetmask)); err != nil {
```

Comment:
```suggestion
	if err := d.Set("hypervisor_netmask", res.HypervisorNetmask); err != nil {
```
brackets are redundant make changes everywhere.